flash/image/face_detection/model.py

```diff
@@ -0,0 +1,192 @@
+
+    def serialize(self, sample: Any) -> Dict[str, Any]:
+        sample = sample[DefaultDataKeys.PREDS] if isinstance(sample, Dict) else sample
+        return sample
+
+
+class FaceDetector(Task):
+    """The ``FaceDetector`` is a :class:`~flash.Task` for detecting faces in images.
+
+    For more details, see
+    :ref:`face_detection`.
+    Args:
+        model: a string of :attr`_models`. Defaults to 'lffd_slim'.
+        pretrained: Whether the model from fastface should be loaded with it's pretrained weights.
+        loss: the function(s) to update the model with. Has no effect for fastface models.
+        metrics: The provided metrics. All metrics here will be logged to progress bar and the respective logger.
+            Changing this argument currently has no effect.
+        optimizer: The optimizer to use for training. Can either be the actual class or the class name.
+        learning_rate: The learning rate to use for training
+    """
+
+    required_extras: str = "image"
+
+    def __init__(
+        self,
+        model: str = "lffd_slim",
+        pretrained: bool = True,
+        loss=None,
+        metrics: Union[Callable, nn.Module, Mapping, Sequence, None] = None,
+        optimizer: Type[Optimizer] = torch.optim.AdamW,
+        learning_rate: float = 1e-4,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+        preprocess: Optional[Preprocess] = None,
+        **kwargs: Any,
+    ):
+        self.save_hyperparameters()
+
+        if model in ff.list_pretrained_models():
+            model = FaceDetector.get_model(model, pretrained, **kwargs)
+        else:
+            ValueError(f"{model} is not supported yet.")
+
+        super().__init__(
+            model=model,
+            loss_fn=loss,
+            metrics=metrics or {"AP": ff.metric.AveragePrecision()},
+            learning_rate=learning_rate,
+            optimizer=optimizer,
+            serializer=serializer or DetectionLabels(),
+            preprocess=preprocess or FaceDetectionPreprocess(),
+        )
+
+    @staticmethod
+    def get_model(
+        model_name,
+        pretrained,
+        **kwargs,
+    ):
+        model, pl_model = FACE_DETECTION_BACKBONES.get(model_name)(pretrained=pretrained, **kwargs)
+
+        # set preprocess params
+        model.register_buffer("normalizer", getattr(pl_model, "normalizer"))
+        model.register_buffer("mean", getattr(pl_model, "mean"))
+        model.register_buffer("std", getattr(pl_model, "std"))
+
+        # set postprocess function
+        setattr(model, "_postprocess", getattr(pl_model, "_postprocess"))
+
+        return model
+
+    def forward(self, x: List[torch.Tensor]) -> Any:
+        images = self._prepare_batch(x)
+        logits = self.model(images)
+
+        # preds: torch.Tensor(B, N, 5)
+        # preds: torch.Tensor(N, 6) as x1,y1,x2,y2,score,batch_idx
+        preds = self.model.logits_to_preds(logits)
+        preds = self.model._postprocess(preds)
```

Comment:
the postprocess wasn't meant to be called within the model directly. This is handled by the DataPipeline.

Reply:
@tchaton from what I understand, this is a call to a postprocess function defined in the fastface library's pl module itself. https://github.com/borhanMorphy/fastface/blob/master/fastface/module.py#L200

So it is called in this manner.

Reply:
Oh my bad. Thanks